Source/Dialogs/Dialog_MathBillConfig.cs

```
@@ -153,192 +153,226 @@ private void RenderMiddlePanel(Rect rect) {
 			Listing_Standard listing_standard = new Listing_Standard();
 			listing_standard.Begin(rect);
 
-			RenderBillSettings(listing_standard);
-			RenderStockpileSettings(listing_standard);
-			RenderWorkerSettings(listing_standard);
-
-			listing_standard.End();
+			try
+			{
+				RenderBillSettings(listing_standard);
+				RenderStockpileSettings(listing_standard);
+				RenderWorkerSettings(listing_standard);
+			}
+			finally
```

I'd like to have some error logging, mainly to prevent confusion when otherwise valid bills fail to resolve.

As it stands this just completely ignores errors, rather than solving them or even notifying the user that their bill may fail to resolve.

Something along these lines would be great--
```cs
//Math.cs
public static List<Exception> loggedExceptions = new List<Exception>();
//...
public static void tryLogException(Exception e) {
  if (loggedExceptions.Contains(e)) return;
  Log.Error("Encountered an error while resolving a bill - logging it once and trying to recover smoothly.")
  Log.Error(e)
  loggedExceptions.Add(e);
}
//...
```
```cs
try {
  //...
} catch (Exception e) {
  Math.tryLogException(e)
} finally {
  //...
}
```

---

I'm not totally sure if there's confusion here or not. I didn't add any catches so the errors will still be thrown and logged.

---

Additionally, since Exceptions are classes and don't seem to override the `.Equals()` method, I'm fairly certain you wouldn't be able to do a one-time log like this.

---

I know the issues are coming from Math, it's been on my todo list to fix this for a while, just haven't been able to touch RimWorld for quite a while because of things going on IRL.  
> I didn't add any catches so the errors will still be thrown [...]  

Ah, I see. That'd log (potentially multiple) exceptions every frame, very quickly filling out the log length limit, which is... not ideal. Doing a one-time log would be much prefered.  

> Since Exceptions are classes that don't seem to override `.Equals()` [...]  

You should be able to, though you might have to `toString` it first. There's a way I've done it in the past, I just can't remember right now for the life of me. Give it a try, and if you can't get it working it's not a big deal I'll make-do.